compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -1620,8 +1620,9 @@ class TreeUnpickler(reader: TastyReader,
         try cs match {
           case ot :: cs1 =>
             if (ot.addr.index == addr.index) {
-              assert(current.exists, i"no symbol at $addr")
-              current
+              // If current does not exist it is beacause this is a pickled expression
+              // and the owner is located outside the expression (i.e. pickled quote).
+              current.orElse(ctx.owner)
```

Comment:
This could hide bugs if we're not in fact in a pickled quote. I think it'd be clearer if we stored the quote owner instead. Something like:
```scala
/** If this is a pickled quote, the owner of the quote, otherwise NoSymbol. */
private var rootOwner: Symbol = NoSymbol

...

def unpickle(mode: UnpickleMode)(using Context): List[Tree] = {
  if mode != UnpickleMode.TopLevel then rootOwner = ctx.owner
  ...
}  
...
class OwnerTree(val addr: Addr, tag: Int, reader: TreeReader, val end: Addr) {
  ...
    try search(children, rootOwner)
  ...
}
```